Sources/web3swift/Operations/WriteOperation.swift

```diff
@@ -11,8 +11,13 @@ import Core
 public class WriteOperation: ReadOperation {
 
     // FIXME: Rewrite this to CodableTransaction
-    public func writeToChain(password: String) async throws -> TransactionSendingResult {
-        await transaction.resolve(provider: web3.provider)
+    /// Sends (raw) transaction for write operation.
+    /// - Parameters:
+    ///   - password: Password for private key.
+    ///   - policies: Custom policies for how to resolve (optional). Default are policies on Web3Provider instance.
+    public func writeToChain(password: String, policies: Policies? = nil) async throws -> TransactionSendingResult {
+        let policiesOrUserDefaults = policies ?? web3.provider.policies
+        try await resolver.resolveAll(for: &transaction, with: policiesOrUserDefaults)
```

Comment:
Think this is a good solution and very flexible. What do you think? @yaroslavyaroslav

Reply:
Question is should this be added to `callContractMethod()` as well? Probably yes?

Reply:
Though this one is great about its intent and implementation, I worry if it a bit magical in terms of default policies applying. LIke user passes nil to that call and gets… default one. Thought it's not critical maybe it'll be a better way to made it non optional and show to user explicitly by method sign, that if they didn't set here anything default will be applied instead.

Reply:
And speaking about your question, I think yes, it's totally worth to add it there as well.

Reply:
I know you wanted `policies: Policies = web3.provider.policies` but the issue is that web3 is a member of the instance, so not accessible as a default here.

So only thing I could think of, is reverting to: `policies: Policies = .auto` but then the whole web3 provider policies is basically unused.

Reply:
Or did you mean just `policies: Policies?` - without a default?

Reply:
And what do you think about this bit:

```swift
class ReadOperation {
    var web3: Web3
    public var policies: Policies { web3.provider.policies }
}

public func writeToChain(password: String, policies: Policies = super.policies) async throws -> TransactionSendingResult {
    try await resolver.resolveAll(for: &transaction, with: policies)
}
```

Or we could bring here public provider instead of policies as well, but I'm not sure if it worth it.